### PR TITLE
Update error messages

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
### Context

The teacher training API currently generates error messages that don’t follow GDS recommendations. Let’s fix that!

### Changes proposed in this pull request

* Capitalise reference number abbreviations (**Unique Reference Number** and **UK Provider Reference Number**)
* Don’t capitalise maths and science in error messages
* Don’t say ‘You must’
* Use ‘Select‘ not ‘Pick’

### Guidance to review

Not so much guidance, but some questions:

* Do tests on Publish need to be updated to expect these revised error messages?
* Do these changes follow the GDS recommendations? – @B-Bonner 

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Review by dankmitchell — APPROVED** [review]

**Review by stevehook — APPROVED** [review]
Looks good.

Just wondering about this one - https://github.com/DFE-Digital/teacher-training-api/blob/master/config/locales/en.yml#L152 - Should it be _Provider name should be 100 characters or fewer_?

File: spec/requests/api/v2/build_new_course_spec.rb
Comment (B-Bonner) [inline spec/requests/api/v2/build_new_course_spec.rb:214]: What's the question that triggers this error message? 

I'm thinking that 'Enter the date when applications open' might be clearer, but I'd need to see how the question's phrased.

Comment (paulrobertlloyd) [reply spec/requests/api/v2/build_new_course_spec.rb:214]: @B-Bonner You can select from two options, one of which allows you to enter a specific date. Tough one…

Screenshot 2021-07-06 at 10 27 36

File: spec/requests/api/v2/providers/courses/update_degree_subject_requirements_spec.rb
Comment (B-Bonner) [inline spec/requests/api/v2/providers/courses/update_degree_subject_requirements_spec.rb:32]: Suggest changing this to: 'Must have an A level in maths'.

File: spec/requests/api/v2/providers/courses/update_degree_subject_requirements_spec.rb
Comment (B-Bonner) [inline spec/requests/api/v2/providers/courses/update_degree_subject_requirements_spec.rb:50]: Suggest changing this to: 'Must have an A level in maths', 'Must have an A level in physics'.

File: spec/controllers/api/v2/courses_controller_spec.rb
Comment (B-Bonner) [inline spec/controllers/api/v2/courses_controller_spec.rb:110]: Change to 'Select if you can sponsor visas'

File: spec/controllers/api/v2/courses_controller_spec.rb
Comment (B-Bonner) [inline spec/controllers/api/v2/courses_controller_spec.rb:73]: Change to 'Select if you can sponsor visas'

File: config/locales/en.yml
Comment (B-Bonner) [inline config/locales/en.yml:123]: Change to 'Select if you can sponsor visas'

File: config/locales/en.yml
Comment (B-Bonner) [inline config/locales/en.yml:93]: What's the question that triggers this? Is it supposed to read 'Select an A level'? Or 'Select a level (of something)'?

Comment (paulrobertlloyd) [reply config/locales/en.yml:93]: Level of education: Primary, Secondary or Further:

Screenshot 2021-07-06 at 10 38 00

File: spec/models/course_spec.rb
Comment (B-Bonner) [inline spec/models/course_spec.rb:301]: Change to 'Select a course level'

File: spec/models/course_spec.rb
Comment (B-Bonner) [inline spec/models/course_spec.rb:321]: Change to 'Select a course level' (if this relates to Primary/Secondary/Further education).

File: spec/requests/api/v2/build_new_course_spec.rb
Comment (B-Bonner) [inline spec/requests/api/v2/build_new_course_spec.rb:270]: Change to 'Select a course level' (if this relates to Primary/Secondary/Further education).

File: spec/requests/api/v2/build_new_course_spec.rb
Comment (B-Bonner) [inline spec/requests/api/v2/build_new_course_spec.rb:372]: Change to 'Select a course level' (if this relates to Primary/Secondary/Further education).

File: spec/requests/api/v2/build_new_course_spec.rb
Comment (B-Bonner) [reply spec/requests/api/v2/build_new_course_spec.rb:214]: I that case, I think:

- If the user doesn't select an option, it should read 'Select when applications will open'.
- If the user selects 'On a specific date', but doesn't enter the date, the error message should read 'Enter the date applications will open'.

Comment (paulrobertlloyd) [reply spec/requests/api/v2/build_new_course_spec.rb:214]: I think, at least for now, we’ll need one message that can cover both scenarios.

File: config/locales/en.yml
Comment (B-Bonner) [reply config/locales/en.yml]: Ok, so yeah, 'Select a course level'

File: spec/requests/api/v2/build_new_course_spec.rb
Comment (B-Bonner) [reply spec/requests/api/v2/build_new_course_spec.rb:214]: Ok, then I think 'Select when applications will open'. Because its a radio button.

**Review by B-Bonner — APPROVED** [review]

File: config/locales/en.yml
Comment (dankmitchell) [inline config/locales/en.yml:124]: please don't merge this yet

File: config/locales/en.yml
Comment (dankmitchell) [inline config/locales/en.yml:125]: or this

File: config/locales/en.yml
Comment (paulrobertlloyd) [reply config/locales/en.yml:124]: 👍 Still trying to get the tests to pass…

Comment (gpeng) [inline config/locales/en.yml:124]: These will break publish so we'll need to update there first to support the old and the new format and then change here. The validation behaviour in publish is unfortunately bound to the string that API returns.

**Review by gpeng — CHANGES_REQUESTED** [review]
This will b

File: config/locales/en.yml
Comment (paulrobertlloyd) [reply config/locales/en.yml:124]: Didn’t this PR address this, or not? https://github.com/DFE-Digital/publish-teacher-training/pull/1792/files

**Comment by paulrobertlloyd** [comment]
@dankmitchell @gpeng This PR is now ready to merge once any dependencies on Publish are sorted.

**Review by gpeng — APPROVED** [review]

**Pull request merged**
